app/main.py

```diff
@@ -1 +1,44 @@
-# write your code here
+from typing import Any
+
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str,
+                 health: int = 100, hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def take_damage(self, damage: int = 50) -> None:
+        self.health -= damage
+        if self.health <= 0:
+            self.die()
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    @classmethod
+    def __str__(cls) -> str:
+        return f"[{', '.join(str(animal) for animal in cls.alive)}]"
```

Comment:
The `__str__` method is typically used for instance-level string representation, not class-level. Consider renaming this method to something more appropriate, like `alive_list_str`, to avoid confusion. Alternatively, you could override the `__repr__` method at the class level to achieve the desired functionality.